File: 11-scrabble-score/scrabble.rb
class Scrabble

  def self.score(word)
    word_array = word.downcase.split("")
    word_array.inject(:+)
  end

  def double
    # letter score *2
  end

  def triple
    # letter score *3
  end
  
  def values
    {
    1 => %w(a e i o u l n r s t)
    2 => %w(d g)
    3 => %w(b c m p)
    4 => %w(f h v w y)
    5 => %w(k)
    8 => %w(k x)
    10 => %w(q z)
    }  
  end

end

File: 11-scrabble-score/scrabble.js
require('./underscore');

global.Scrabble = function(){};

Scrabble = {
	score: function(word){
		word = word || ""
		var letters = word.split("")
		//var totalScores = _.map(letters, function(letter){ 
			//this.letterScores(letters[i]); 
		//});
		var currentScore = 0;
		for(i = 0; i <= word.length-1; i++) {
			currentScore += this.letterScores()[word[i]];
		}
		return currentScore;
	},
	letterScores: function() {
	 return { 
	  	a : 1, b : 3, c : 3, d : 2, e : 1,
			f : 4, g : 2, h : 4, i : 1, j : 8,
			k : 5, l : 1, m : 3, n : 1, o : 1,
			p : 3, q : 10, r : 1, s : 1, t : 1,
		  u : 1, v : 4, w : 4, x : 8, y : 4,
      z : 10 
    }
	}
}
// word.split("").map{|l|values[l]}.inject(:+)

// _.map([1, 2, 3], function(num){ return num * 3; });